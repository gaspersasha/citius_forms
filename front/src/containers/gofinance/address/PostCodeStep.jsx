import React from 'react';
import PropTypes from 'prop-types';
import { LoqateAddress } from '~containers';
import s from './address.module.sass';

const PostCodeStep = ({
  changeAddressField,
  changePostcodeField,
  selectAddressHandler,
  postCode,
  houseName,
  houseNumber,
  street,
  district,
  town,
  county,
}) => (
  <div className={s.addressFormWrapper}>
    <LoqateAddress
      selectHandler={selectAddressHandler}
      formName="address"
      subTitle="Start typing your address and select the relevant option when it appears in the list below. If it is not found then click on the 'Enter address manually' link"
      manualFields={[
        {
          ...postCode,
          label: 'Postcode',
          name: 'postCode',
          required: true,
          onChange: changePostcodeField,
          warning: '* Please type correct UK postal code',
        },
        {
          ...houseName,
          label: 'House name',
          name: 'houseName',
          onChange: changeAddressField,
        },
        {
          ...houseNumber,
          label: 'House number',
          name: 'houseNumber',
          onChange: changeAddressField,
        },
        {
          ...street,
          label: 'Street',
          name: 'street',
          onChange: changeAddressField,
          required: true,
        },
        {
          ...district,
          label: 'District',
          name: 'district',
          onChange: changeAddressField,
        },
        {
          ...town,
          label: 'Town',
          name: 'town',
          onChange: changeAddressField,
          required: true,
        },
        {
          ...county,
          label: 'County',
          name: 'county',
          onChange: changeAddressField,
        },
      ]}
    />
  </div>
);

const addressType = {
  value: PropTypes.string,
  status: PropTypes.string,
};

PostCodeStep.propTypes = {
  selectAddressHandler: PropTypes.func.isRequired,
  changePostcodeField: PropTypes.func.isRequired,
  changeAddressField: PropTypes.func.isRequired,
  postCode: PropTypes.shape(addressType).isRequired,
  houseName: PropTypes.shape(addressType).isRequired,
  houseNumber: PropTypes.shape(addressType).isRequired,
  street: PropTypes.shape(addressType).isRequired,
  district: PropTypes.shape(addressType).isRequired,
  town: PropTypes.shape(addressType).isRequired,
  county: PropTypes.shape(addressType).isRequired,
};

export default PostCodeStep;
