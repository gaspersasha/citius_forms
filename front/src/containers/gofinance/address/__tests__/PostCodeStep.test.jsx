import React from 'react';
import { INPUT_STATUS, VALIDATION_TYPES } from '~constants';
import PostCodeStep from '../PostCodeStep';

describe('<PostCodeStep />', () => {
  const addressMock = {
    value: 'qwerty',
    status: INPUT_STATUS.VALID,
    validationType: VALIDATION_TYPES.NONE,
  };

  const props = {
    selectAddressHandler: jest.fn(),
    changePostcodeField: jest.fn(),
    changeAddressField: jest.fn(),
    postCode: addressMock,
    houseName: addressMock,
    houseNumber: addressMock,
    street: addressMock,
    district: addressMock,
    town: addressMock,
    county: addressMock,
  };

  const wrapper = shallow(<PostCodeStep {...props} />);

  it('renders without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
