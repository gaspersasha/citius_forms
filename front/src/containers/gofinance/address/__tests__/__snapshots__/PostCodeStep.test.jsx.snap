// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCodeStep /> renders without errors 1`] = `
<div
  className="addressFormWrapper"
>
  <LoqateAddress
    errorMessage=""
    formName="address"
    manualFields={
      Array [
        Object {
          "label": "Postcode",
          "name": "postCode",
          "onChange": [MockFunction],
          "required": true,
          "status": "valid",
          "validationType": "none",
          "value": "qwerty",
          "warning": "* Please type correct UK postal code",
        },
        Object {
          "label": "House name",
          "name": "houseName",
          "onChange": [MockFunction],
          "status": "valid",
          "validationType": "none",
          "value": "qwerty",
        },
        Object {
          "label": "House number",
          "name": "houseNumber",
          "onChange": [MockFunction],
          "status": "valid",
          "validationType": "none",
          "value": "qwerty",
        },
        Object {
          "label": "Street",
          "name": "street",
          "onChange": [MockFunction],
          "required": true,
          "status": "valid",
          "validationType": "none",
          "value": "qwerty",
        },
        Object {
          "label": "District",
          "name": "district",
          "onChange": [MockFunction],
          "status": "valid",
          "validationType": "none",
          "value": "qwerty",
        },
        Object {
          "label": "Town",
          "name": "town",
          "onChange": [MockFunction],
          "required": true,
          "status": "valid",
          "validationType": "none",
          "value": "qwerty",
        },
        Object {
          "label": "County",
          "name": "county",
          "onChange": [MockFunction],
          "status": "valid",
          "validationType": "none",
          "value": "qwerty",
        },
      ]
    }
    selectHandler={[MockFunction]}
    subTitle="Start typing your address and select the relevant option when it appears in the list below. If it is not found then click on the 'Enter address manually' link"
  />
</div>
`;
